stratosphere/ams_mitm/source/set_mitm/setsys_mitm_service.cpp

```diff
@@ -102,6 +184,44 @@ namespace ams::mitm::settings {
         return GetFirmwareVersionImpl(out.GetPointer(), m_client_info);
     }
 
+    Result SetSysMitmService::SetBluetoothDevicesSettings(const sf::InMapAliasArray<SetSysBluetoothDevicesSettings> &settings) {
+        /* Forward to session unless external database setting enabled. */
+        R_UNLESS(ExternalBluetoothDatabaseEnabled(), sm::mitm::ResultShouldForwardToSession());
+
+        /* Create the external database if it doesn't exist. */
+        if (!HasExternalBluetoothDatabase()) {
+            R_TRY(fs::CreateFile(g_external_bluetooth_db_location, 0));
+        }
+
+        /* Backup local database to sd card. */
+        R_TRY(WriteExternalBluetoothDatabase(settings.GetPointer(), settings.GetSize()));
+
+        /* Also allow the updated database to be stored to system save as usual. */
+        return sm::mitm::ResultShouldForwardToSession();
```

Comment:
This makes me uncomfortable.

Please make it a forwarding command so that the function ends in R_TRY(); return ResultSuccess(); ?